### PR TITLE
Issue Fixes

### PR DESCRIPTION
**Pull request by namithj**
# Pull Request

## What changed?

[#56](https://github.com/aspirepress/AspireUpdate/issues/56)
[#47](https://github.com/aspirepress/AspireUpdate/issues/47)
[#43](https://github.com/aspirepress/AspireUpdate/issues/43)
[#42](https://github.com/aspirepress/AspireUpdate/issues/42)
[#41](https://github.com/aspirepress/AspireUpdate/issues/41)

## Why did it change?

Issues Fixed

## Did you fix any specific issues?

[#56](https://github.com/aspirepress/AspireUpdate/issues/56)
[#47](https://github.com/aspirepress/AspireUpdate/issues/47)
[#43](https://github.com/aspirepress/AspireUpdate/issues/43)
[#42](https://github.com/aspirepress/AspireUpdate/issues/42)
[#41](https://github.com/aspirepress/AspireUpdate/issues/41)

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

File: includes/class-debug.php
Comment (asirota) [inline includes/class-debug.php:1]: Would be nicer to have more friendly error comment such as

"Could not open or write to the host file system."

**Review by asirota — CHANGES_REQUESTED** [review]
Check the error message language for WordPress_filesystem

**Comment by asirota** [comment]
@namithj can you add 

`Closes: `

Prefix before each issue number link in the fixed section.

File: includes/class-debug.php
Comment (namithj) [reply includes/class-debug.php:1]: Can we use something else in place of Hosts.  We are already using Host as in the API Host.

Comment (namithj) [reply includes/class-debug.php:1]: Added to file

**Review by asirota — APPROVED** [review]

**Comment by costdev** [comment]
Reviewing is much harder for contributors when they have to learn and follow multiple contexts.

Is it okay if we avoid solving multiple unrelated issues in one PR?

**Comment by asirota** [comment]
So then one issue and one pr per issue? @namithj what do you think of that?

**Review by asirota — COMMENTED** [review]
Looks good

**Comment by namithj** [comment]
It's alright for me.  This is more of a time saving measure since we only have 6-7 files in the plugin to take care of

**Comment by asirota** [comment]
Let's try to make the Pr more atomic to one or two related issues tougher then.

**Pull request merged**